crates/oxc_ast/src/serialize/mod.rs

```diff
@@ -168,10 +168,14 @@ impl ESTree for ProgramConverter<'_, '_> {
         let span_start =
             if S::INCLUDE_TS_FIELDS { get_ts_start_span(program) } else { program.span.start };
 
+        let range = serializer.range();
         let mut state = serializer.serialize_struct();
         state.serialize_field("type", &JsonSafeString("Program"));
         state.serialize_field("start", &span_start);
         state.serialize_field("end", &program.span.end);
+        if range {
+            state.serialize_field("range", &[span_start, program.span.end]);
+        }
```

Comment:
If `range` is last field (which is what you have in the codegen), it should be last here too, for consistency.